crates/sui-config/src/node.rs

```diff
@@ -71,6 +73,11 @@ pub struct NodeConfig {
     #[serde(default)]
     pub enable_reconfig: bool,
 
+    /// Number of checkpoints per epoch. Exposing this in config to allow easier testing.
+    /// It will be removed down the road.
+    #[serde(default = "default_checkpoints_per_epoch")]
+    pub checkpoints_per_epoch: u64,
```

Comment:
I do not think that this is the correct place for such configuration. This is a value that all the validators are going to need to agree upon in order for proper operations of the chain. As such the only appropriate places for this are going to either be hard-coded in the binary or as an on-chain configuration value in the system object.

I know that you have a comment saying it'll be removed, but i think we should be thinking now about how such values should be maintained in our system.

Reply:
Actually we probably should gate this as an Option<u64> so that None indicates, never do reconfig so that devnet wont break when this is deployed and we haven't fixed all the bugs.

Reply:
There is a separate enable_reconfig flag, but I guess perhaps I should merge them

Reply:
ah i didn't know that, maybe? i'll leave that up to you :)